moveit_ros/move_group/src/default_capabilities/cartesian_path_service_capability.cpp

```diff
@@ -149,7 +149,7 @@ bool MoveGroupCartesianPathService::computeService(
             constraint_fn = boost::bind(
                 &isStateValid,
                 req->avoid_collisions ? static_cast<const planning_scene::PlanningSceneConstPtr&>(*ls).get() : nullptr,
-                kset->empty() ? nullptr : kset.get(), _1, _2, _3);
+                kset->empty() ? nullptr : kset.get(), boost::placeholders::_1, boost::placeholders::_2, boost::placeholders::_3);
```

Comment:
broader question: why are we using boost::bind instead of std::bind? Is there a lower-level library requiring it? If not, seems like we should update MoveIt2 to reduce our boost usage (core ROS2 has done so).

Reply:
... or [use lambda's](https://github.com/peter-can-write/cpp-notes/blob/master/prefer-lambdas-over-std::bind.md).

Reply:
The only downside to lambdas is that they aren't as broadly understood - they seem a little more "advanced". In various discussions I've recently had with ROS/ROS2 package maintainers, I've noticed there is pretty decent evidence that the more advanced the codebase, the harder it is to get community help.

Reply:
I agree that we should switch to lambdas in cases where the function is trivial (i.e. log message and return of a result) or where another member function should be called but we need to preprocess the arguments. Otherwise,`std::bind` is often much shorter and readable.